code/engine/xrCore/_math.cpp

```diff
@@ -201,92 +199,52 @@ void _initialize_cpu() {
-    FPU::m24r();
-#endif // XRCORE_STATIC
-    if (CPU::ID.hasFeature(CpuFeature::Sse)) {
-        //_mm_setcsr ( _mm_getcsr() | (_MM_FLUSH_ZERO_ON+_MM_DENORMALS_ZERO_ON) );
-        _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
-        if (_denormals_are_zero_supported) {
-            __try {
-                _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
-            } __except (EXCEPTION_EXECUTE_HANDLER) {
-                _denormals_are_zero_supported = FALSE;
-            }
-        }
-    }
-}
+#ifdef _WIN32
+const DWORD MS_VC_EXCEPTION = 0x406D1388;
 
-// threading API
 #pragma pack(push, 8)
-struct THREAD_NAME {
-    DWORD dwType;
-    LPCSTR szName;
-    DWORD dwThreadID;
-    DWORD dwFlags;
-};
-void thread_name(const char* name) {
-    THREAD_NAME tn;
-    tn.dwType = 0x1000;
-    tn.szName = name;
-    tn.dwThreadID = DWORD(-1);
-    tn.dwFlags = 0;
+typedef struct tagTHREADNAME_INFO {
+    DWORD dwType;     // Must be 0x1000.
+    LPCSTR szName;    // Pointer to name (in user addr space).
+    DWORD dwThreadID; // Thread ID (-1=caller thread).
+    DWORD dwFlags;    // Reserved for future use, must be zero.
+} THREADNAME_INFO;
+#pragma pack(pop)
+
+void set_thread_name(uint32_t dwThreadID, const char* threadName) {
+    // DWORD dwThreadID = ::GetThreadId( static_cast<HANDLE>( t.native_handle() ) );
+
+    THREADNAME_INFO info;
+    info.dwType = 0x1000;
+    info.szName = threadName;
+    info.dwThreadID = dwThreadID;
+    info.dwFlags = 0;
+
     __try {
-        RaiseException(0x406D1388, 0, sizeof(tn) / sizeof(DWORD), (ULONG_PTR*)&tn);
-    } __except (EXCEPTION_CONTINUE_EXECUTION) {
+        RaiseException(MS_VC_EXCEPTION, 0, sizeof(info) / sizeof(ULONG_PTR), (ULONG_PTR*)&info);
+    } __except (EXCEPTION_EXECUTE_HANDLER) {
     }
 }
-#pragma pack(pop)
+void set_thread_name(const char* threadName) { set_thread_name(GetCurrentThreadId(), threadName); }
 
-struct THREAD_STARTUP {
-    thread_t* entry;
-    char* name;
-    void* args;
-};
-void __cdecl thread_entry(void* _params) {
-    // initialize
-    THREAD_STARTUP* startup = (THREAD_STARTUP*)_params;
-    thread_name(startup->name);
-    thread_t* entry = startup->entry;
-    void* arglist = startup->args;
-    xr_delete(startup);
-    _initialize_cpu_thread();
-
-    // call
-    entry(arglist);
+void set_thread_name(const char* threadName, std::thread* thread) {
+    DWORD threadId = ::GetThreadId(static_cast<HANDLE>(thread->native_handle()));
+    set_thread_name(threadId, threadName);
 }
 
-void thread_spawn(thread_t* entry, const char* name, unsigned stack, void* arglist) {
-    Debug._initialize();
-
-    THREAD_STARTUP* startup = xr_new<THREAD_STARTUP>();
-    startup->entry = entry;
-    startup->name = (char*)name;
-    startup->args = arglist;
-    _beginthread(thread_entry, stack, startup);
+#else
+void set_thread_name(const char* threadName, std::thread* thread) {
+    auto handle = thread->native_handle();
+    pthread_setname_np(handle, threadName);
```

Comment:
Are you sure this is working? Official ways is SetThreadDescription and the method above, which is described [here](https://docs.microsoft.com/ru-ru/visualstudio/debugger/how-to-set-a-thread-name-in-native-code)

Reply:
this is not for windows